types/types.go

```diff
@@ -91,14 +91,29 @@ type ImageReference interface {
 	DeleteImage(ctx context.Context, sys *SystemContext) error
 }
 
+// LayerCompression indicates if layers must be compressed, decompressed or preserved
+type LayerCompression int
+
+const (
+	// PreserveOriginal indicates the layer must be preserved, ie
+	// no compression or decompression.
+	PreserveOriginal LayerCompression = iota
+	// Decompress indicates the layer must be decompressed
+	Decompress
+	// Compress indicates the layer must be compressed
+	Compress
+)
+
 // BlobInfo collects known information about a blob (layer/config).
 // In some situations, some fields may be unknown, in others they may be mandatory; documenting an “unknown” value here does not override that.
 type BlobInfo struct {
-	Digest      digest.Digest // "" if unknown.
-	Size        int64         // -1 if unknown
-	URLs        []string
-	Annotations map[string]string
-	MediaType   string
+	Digest               digest.Digest // "" if unknown.
+	Size                 int64         // -1 if unknown
+	URLs                 []string
+	Annotations          map[string]string
+	MediaType            string
+	Compression          LayerCompression
+	CompressionAlgorithm string // as defined pkg/compression.{GzipCompression, ZstdCompression, ...}
```

Comment:
Can’t this be the native `compression.Algorithm` type directly? (I guess that runs into visibility issues?)

Reply:
That should be possible and I prefer your suggestion (in combination with a "don't edit" switch).